### PR TITLE
feat: early remote profile resolution

### PR DESCRIPTION
**Pull request by Jakub-Vacek**
This PR add ability to load profile remotely without CLI installation. It moves profile resolution to `getProfile` method and adds check to ensure that use case exists in profile AST.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have updated the documentation accordingly. If you are changing **code related to user secrets** you need to really make sure that [security documentation](https://github.com/superfaceai/one-sdk-js/blob/main/SECURITY.md) is correct.
- [x] I have read the [CONTRIBUTING](https://github.com/superfaceai/one-sdk-js/blob/main/CONTRIBUTING.md) document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by TheEdward162 — APPROVED** [review]

File: src/core/profile-provider/profile-provider.ts [hunk 1]
```diff
@@ -46,7 +43,7 @@ import { resolveSecurityConfiguration } from './security';
 const DEBUG_NAMESPACE = 'profile-provider';
 
 export async function bindProfileProvider(
-  profileConfig: ProfileConfiguration,
+  profile: ProfileBase,
```
[the remark this answers is not in the document]
Comment (lukas-valenta) [reply src/core/profile-provider/profile-provider.ts:46]: but leave it for now, should use IProfile from #258

File: src/core/client/client.internal.ts [hunk 2]
```diff
@@ -28,11 +40,110 @@ export class InternalClient {
...
+   * File property:
+   *  - loads directly passed file
+   *  - can point only to .supr.ast.json file
+   *  - throws if file not found or not valid ProfileDocumentNode
+   * Version property:
+   *  - looks for [profileId]@[version].supr.ast.json file in superface/grid
+   *  - if not found it tries to fetch profile AST from Registry
+   * @returns ProfileDocumentNode
+   */
+  public async resolveProfileAst(
```
Comment (lukas-valenta) [inline src/core/client/client.internal.ts:54]: question: does this need to be a method on the client?

File: src/core/profile-provider/profile-provider.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (lukas-valenta) [reply src/core/profile-provider/profile-provider.ts:46]: third iteration: just use what you need, which is AST?

**Review by lukas-valenta — CHANGES_REQUESTED** [review]

File: src/core/client/client.internal.ts (hunk 2 above)
Comment (Jakub-Vacek) [reply src/core/client/client.internal.ts:54]: Probably not. I wasn't sure where to move it.

File: src/core/client/client.internal.ts [hunk 3]
```diff
@@ -28,11 +40,110 @@ export class InternalClient {
...
+        'Reading of possible profile file failed with error %O',
+        error
+      );
+      void error;
+    }
+
+    logFunction?.('Fetching profile file from registry');
+
+    // Fallback to remote
+    // TODO: cache this somewhere (similar to CLI - .cache, config.cachePath, in memory)?
```
[the remark this answers is not in the document]
Comment (Jakub-Vacek) [reply src/core/client/client.internal.ts:119]: When I think about it I think we should leave caching AST (and looking for it) to another PR. Cache position should be probably discussed little bit more with more people.

File: src/core/client/client.internal.ts (hunk 2 above)
Comment (Jakub-Vacek) [reply src/core/client/client.internal.ts:55]: @lukas-valenta should I move it? 😅

Comment (lukas-valenta) [reply src/core/client/client.internal.ts:55]: yeah, move it to a function near Profile

File: src/core/profile/resolve-profile-ast.test.ts [hunk 4]
```diff
@@ -0,0 +1,449 @@
...
+
+// const mockSuperJsonCustomPath = new SuperJson({
+//   profiles: {
+//     test: '2.1.0',
+//   },
+//   providers: {
+//     quz: {},
+//   },
+// });
+
```
Comment (lukas-valenta) [inline src/core/profile/resolve-profile-ast.test.ts:60]:
```suggestion
```

**Review by lukas-valenta — APPROVED** [review]

**Pull request merged**